src/main/com/fulcrologic/fulcro/rendering/ident_optimized_render.cljc

```diff
@@ -108,15 +103,12 @@
           (render-dependents-of-ident! app i)))
       (let [state-map          @state-atom
             idents-in-joins    (or idents-in-joins #{})
-            dirty-linked-props (reduce
-                                 (fn [acc p]
-                                   (if (not (identical?
-                                              (get state-map p)
-                                              (get last-rendered-state p)))
-                                     (conj acc p)
-                                     acc))
-                                 #{}
-                                 linked-props)
+            dirty-linked-props (into #{}
+                                     (filter (fn [p]
+                                               (not (identical?
+                                                      (get state-map p)
+                                                      (get last-rendered-state p)))))
+                                     linked-props)
```

Comment:
This is technically an improvement, but again realize that the number of linked props should be relatively small for any given render (users don't change much per transaction usually).  To be honest, I'm not even sure the ident-optimized render ends up "winning" over plain keyframe in a lot of cases...esp if there *were* a lot of dirty props.  It is entirely possible that the *real* optimization in this system is a heuristic that can guess when a keyframe render is going to win before doing all of this other work.

Reply:
I am still learning Fulcro at the moment. Once I am done reading the full documentation and actually use it, I would be happy to contribute to that type of profiling.